emulatorview/src/main/java/jackpal/androidterm/emulatorview/PaintRenderer.java

```diff
@@ -107,7 +107,11 @@ public void drawTextRun(Canvas canvas, float x, float y, int lineOffset,
                     canvas.drawText(text, index, countBeforeCursor, left, textOriginY, mTextPaint);
                 }
                 // Text at cursor
-                mTextPaint.setColor(mPalette[TextStyle.ciCursorForeground]);
+                // Only use invert color with BLOCK cursor.
+                // For underdash and vertical line, use normal color.
+                if(mCursorStyle == 0) {
```

Comment:
Beside this location, there is no other external use of `mCursorStyle`, it might be slightly better to keep it private and access via an accessor ( for a small potential performance cost, but better isolation ).